File: app/src/main/java/me/eagzzycsl/darkroom/access/MachineState.java
package me.eagzzycsl.darkroom.access;


import java.util.HashMap;


public class MachineState {
    private final HashMap<ActivityChange, MachineState> next = new HashMap<>();
    private final String name;
    public MachineState(String name) {
        this.name =name;
    }
    public String getName(){
        return this.name;
    }
    public void setNext(ActivityChange activityChange, MachineState machineState) {
        this.next.put(activityChange, machineState);
    }

    public MachineState accept(ActivityChange activityChange) {
        return this.next.get(activityChange);
    }
}

File: app/src/main/java/me/eagzzycsl/darkroom/state/MachineState.kt
package me.eagzzycsl.darkroom.state


import java.util.HashMap


class MachineState(val name: String) {
    private val next = HashMap<WindowChange, MachineState>()
    fun setNext(windowChange: WindowChange, machineState: MachineState) {
        this.next.put(windowChange, machineState)
    }

    fun accept(windowChange: WindowChange): MachineState? {
        return this.next[windowChange]
    }
}